### ISSUE TITLE
csv-table directive can't handle non-ascii file path

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1512]
This issue was posted to Sphinx-users.jp ML:  http://www.python.jp/pipermail/sphinx-users/2014-February/000937.html

`
.. csv-table:: <non-ascii-path-to-csv-file>
`

This reST raise UnicodeDecodeError. The below path will correct this issue.

```
--- a/sphinx/environment.py     Fri Feb 07 13:35:54 2014 +0900
+++ b/sphinx/environment.py     Fri Feb 07 14:19:27 2014 +0900
@@ -753,7 +753,7 @@

     def process_dependencies(self, docname, doctree):
         """Process docutils-generated dependency info."""
-        cwd = os.getcwd()
+        cwd = os.getcwdu()
         frompath = path.join(path.normpath(self.srcdir), 'dummy')
         deps = doctree.settings.record_dependencies
         if not deps:
```

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/1512
- Originally reported by: Takayuki Shimizukawa
- Originally created at: 2014-07-17T08:12:56.814

**Comment by shimizukawa** [comment]
_From Takeshi Komiya on 2014-08-23 08:17:25+00:00_

I investigate about this problem, it is caused from current working directory, not path of csv file.

In spite of the document does not include any non-ascii paths,
this bug crashes `sphinx-build` if you invoke the command _at non-ascii directory_ ::

```
$ cd <non-ascii-directory>
$ sphinx-build -E -T -b html -d ../doc/_build/doctrees  ../doc ../doc/_build/html
Running Sphinx v1.2.2
loading translations [ja]... done
building [html]: targets for 4 source files that are out of date
updating environment: 4 added, 0 changed, 0 removed
../doc/tox.inis... [ 25%] index

Traceback (most recent call last):
  File "/Users/tkomiya/.buildout/buildout-eggs/Sphinx-1.2.2-py2.7.egg/sphinx/cmdline.py", line 254, in main
    app.build(force_all, filenames)
  File "/Users/tkomiya/.buildout/buildout-eggs/Sphinx-1.2.2-py2.7.egg/sphinx/application.py", line 212, in build
    self.builder.build_update()
  File "/Users/tkomiya/.buildout/buildout-eggs/Sphinx-1.2.2-py2.7.egg/sphinx/builders/__init__.py", line 214, in build_update
    'out of date' % len(to_build))
  File "/Users/tkomiya/.buildout/buildout-eggs/Sphinx-1.2.2-py2.7.egg/sphinx/builders/__init__.py", line 234, in build
    purple, length):
  File "/Users/tkomiya/.buildout/buildout-eggs/Sphinx-1.2.2-py2.7.egg/sphinx/builders/__init__.py", line 134, in status_iterator
    for item in iterable:
  File "/Users/tkomiya/.buildout/buildout-eggs/Sphinx-1.2.2-py2.7.egg/sphinx/environment.py", line 477, in update_generator
    self.read_doc(docname, app=app)
  File "/Users/tkomiya/.buildout/buildout-eggs/Sphinx-1.2.2-py2.7.egg/sphinx/environment.py", line 629, in read_doc
    self.process_dependencies(docname, doctree)
  File "/Users/tkomiya/.buildout/buildout-eggs/Sphinx-1.2.2-py2.7.egg/sphinx/environment.py", line 769, in process_dependencies
    path.normpath(path.join(cwd, dep)))
  File "/Users/tkomiya/.pyenv/versions/2.7.7/lib/python2.7/posixpath.py", line 80, in join
    path += '/' + b
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe3 in position 24: ordinal not in range(128)

Encoding error:
'ascii' codec can't decode byte 0xe3 in position 24: ordinal not in range(128)
The full traceback has been saved in /var/folders/2s/79pf79d94r9gxhzl4kpztrr00000gp/T/sphinx-err-jSa7gf.log, if you want to report the issue to the developers.
```

**Comment by shimizukawa** [comment]
_From Takeshi Komiya on 2014-09-15 11:49:22+00:00_

sorry, my investigating is incorrect. it's a misunderstanding.

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2014-09-23 02:43:18+00:00_

Merged in tk0miya/sphinx/stable (https://bitbucket.org/birkenfeld/sphinx/pull-request/288)

Fix #1512 env.record_dependency crashes on multibyte directories

→ <<cset 5b5937d057770b4fb1f8c2cbe5f719ae1263f7c2>>

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2014-09-23 02:43:18+00:00_

Merged in tk0miya/sphinx/stable (https://bitbucket.org/birkenfeld/sphinx/pull-request/288)

Fix #1512 env.record_dependency crashes on multibyte directories

→ <<cset 5b5937d057770b4fb1f8c2cbe5f719ae1263f7c2>>

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2014-09-23 02:43:18+00:00_

Fix #1512 env.record_dependency crashes on multibyte directories

→ <<cset 873efc53580b04c4f4a6bfeb9d3164cbb2dfbb7b>>
